### ISSUE TITLE
Latest 4.2.0 build on Nuget causes issues because it requires version Npgsql 5.0.3, as opposed to any greater version

### ISSUE THREAD

**Issue by abelbraaksma** [Zaid-Ajaj/Npgsql.FSharp#117]
The most recent version of Npgsql.FSharp has a fixed dependency on Npgsql = 5.0.3, as opposed ot Npgsql >= 5.0.3 (which is what it was before). This causes compile errors or warnings, depending on your config. Local solution is to specifically reference a higher version of `Npgsql`, though this'll still cause warnings, like such:

> warning NU1608: Detected package version outside of dependency constraint: Npgsql.FSharp 4.2.0 requires Npgsql (= 5.0.3) but version Npgsql 6.0.4 was resolved.

Compare 4.2.0:
![image](https://user-images.githubusercontent.com/16015770/168570827-374e2a73-70be-4907-9b14-d1a9533367ba.png)

With 4.1.0:
![image](https://user-images.githubusercontent.com/16015770/168570861-7fc3405a-4c63-4ab9-8713-b13469786fc8.png)

Btw, love the lib! I'll check if I can find where this dependency went wrong and issue a PR.

**Comment by Zaid-Ajaj** [comment]
hi @abelbraaksma, thanks for filing the issue 🙏 

> Which suggests that there's been a change in how Paket resolves the references, as opposed to a source change.

I think you are right, I need to change the version constraint into `~> 5.0.3` instead of exactly `5.0.3`

**Comment by abelbraaksma** [comment]
Thanks for getting back at me so quickly, @Zaid-Ajaj! Strangely, in the `lock` files there doesn't seem to be any difference in syntax. The only thing that stood out to me was that `Npgsql` was removed from `dependencies.lock`.

**Comment by thomasd3** [comment]
@Zaid-Ajaj do you know when you plan to update the nuget?

**Comment by Zaid-Ajaj** [comment]
@thomasd3 As soon as I can, I am really low on free time these days :/

**Comment by Zaid-Ajaj** [comment]
Npgsql.FSharp v5.0.1 has been published and now depends on latest stable version of Npgsql v6.x I believe this resolve this issue

**Comment by abelbraaksma** [comment]
@Zaid-Ajaj, thanks for this! Yes, it is an improvement, but now `Ply` is locked to version `0.3.1`, which I believe previously wasn't the case:

current:
![image](https://user-images.githubusercontent.com/16015770/174067699-b243c7e6-9815-4b9c-86c1-d0de2eff2801.png)

Previously (4.1.0 and before)
![image](https://user-images.githubusercontent.com/16015770/174067785-f5073f89-01fe-4848-856f-b85f25f4a68b.png)

**Comment by thomasd3** [comment]
I'm getting a crash when using the last version:

---> System.TypeLoadException: Could not load type 'System.Data.Common.DbBatch' from assembly 'Npgsql, Version=6.0.4.0, Culture=neutral, PublicKeyToken=5d8b90d52f46fda7'.
   at Ply<void> Npgsql.FSharp.Sql+body@365-13.Invoke()
   at Ply<u> Ply.TplPrimitives.tryFinally<u>(FSharpFunc<void, Ply<u>> continuation, FSharpFunc<void, void> finallyBody)
   at Ply<void> Npgsql.FSharp.Sql+executeTransactionAsync@346-5.Invoke()

**Comment by abelbraaksma** [comment]
@thomasd3, yes, that’s what @yreynhout reported in #121, hence me linking there. Though I’m not convinced they’re related per se, as the exception you got has a different cause (see other issue).
